### ISSUE TITLE
[MNG-6022] mvn.cmd fails if directory contains an ampersand (&)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7843]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** opened **[MNG-6022](https://issues.apache.org/jira/browse/MNG-6022?redirect=false)** and commented

Checkout a fresh version of MASSEMBLY trunk, run `mvn clean install -Prun-its` and see the IT `projects\basic-features\this & that`. Go to the `target` directory of this IT and run Maven again. It says:

```
D:\Entwicklung\Projekte\maven-assembly-plugin\target\it\projects\basic-features\this & that>mvn compile
Der Befehl "that" ist entweder falsch geschrieben oder
konnte nicht gefunden werden.
Das System kann den angegebenen Pfad nicht finden.
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.060 s
[INFO] Finished at: 2016-05-15T00:37:27+02:00
[INFO] Final Memory: 6M/241M
[INFO] ------------------------------------------------------------------------
[ERROR] The goal you specified requires a project to execute but there is no POM in this directory (D:\). Please verify you invoked Maven from the correct directory. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MissingProjectException

D:\Entwicklung\Projekte\maven-assembly-plugin\target\it\projects\basic-features\this & that>
```

`EXEC_DIR` and friends aren't quoted and therefore the ampersand toolks for `that` command.

---

**Affects:** 3.3.9

**Issue Links:**
- [MNG-5962](https://issues.apache.org/jira/browse/MNG-5962) mvn.cmd fails when the current directory has spaces in between
 (_**"depends upon"**_)

**Backported to:** [3.5.0-alpha-1](https://github.com/apache/maven/milestone/92?closed=1)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Implicitly fixed with [a035d9026f45518f430dace7ba930835d0d3e85a](https://git-wip-us.apache.org/repos/asf?p=maven.git;a=commit;h=a035d9026f45518f430dace7ba930835d0d3e85a).

**Comment by jira-importer** [comment]
**[Stephen Connolly](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=stephenc)** commented

Maven 3.4.0 has been dropped. See [this thread](https://mail-archives.apache.org/mod_mbox/maven-dev/201701.mbox/%3CCA%2BnPnMx-e7kGYy3Hp87v8hLGdhp1q%3DtKLx_6QuZ4kGUqHEBGcw%40mail.gmail.com%3E) for more details.

This issue will need to be re-scheduled for a Maven release in the (hopefully near) future.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Christian, do you mind if I reassign this to me? I want to consolidate all start script related issues.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Implicitly fixed with [842b07be97fdb7d4f36e167f596eb185f9601169](https://git-wip-us.apache.org/repos/asf?p=maven.git;a=commit;h=842b07be97fdb7d4f36e167f596eb185f9601169).
